app/helpers/search_helper.rb

```diff
@@ -43,4 +43,9 @@ def comment_or_answer?(searchable)
   def talk?(searchable)
     searchable.instance_of?(User) && searchable.talk.present?
   end
+
+  def can_be_wip?(searchable)
+    class_names = [Announcement, Event, Page, Product, Question, Report]
+    class_names.any? { |class_name| searchable.is_a?(class_name) }
```

Comment:
サブクラスかどうかを考慮しなくてよい (= `is_a?` じゃなくてよい) のであれば以下のように書くとスッキリしそうです。https://api.rubyonrails.org/classes/Object.html#method-i-in-3F

```suggestion
    searchable.class.in?([Announcement, Event, Page, Product, Question, Report])
```

Reply:
wip にできるかどうかが `wip?` みたいな predicate method によって決められるなら以下のような書き方もアリかもです。これだと wip にできるクラスを列挙しなくても済みそう。

```suggestion
    searchable.respond_to?(:wip?)
```

Reply:
ご確認ありがとうございます！流麗！！
`searchable.respond_to?(:wip)`でやっていることがひと目でわかり、メソッドに切り出す必要がなさそうなのでjbuilderに直接書く形に変更してみました。